### ISSUE TITLE
Give error if children array length is zero.

### ISSUE THREAD

**Issue by mumairofficial** [leovo2708/ngx-treeview#48]
I am calling an API and get data in response like this

```json
[{
	"id": 6,
	"text": "System",
	"children": [{
		"id": 3,
		"text": "value 1",
		"children": [] <= error
	}, {
		"id": 7,
		"text": "value 2",
		"children": [] <= error
	}, {
		"id": 8,
		"text": "value 3",
		"children": [] <= error
	}]
},
...]
```

and I process the data like following:

```ts

public items: TreeviewItem[] = [];

private allowedItems(): void {
    this.service.getitemsList().subscribe(
      data => this.processData(data)
    );
}

private processData(data): void {
    data.forEach(element => {
      this.items.push((new TreeviewItem(element)));
    });
}

```

And it generates error like this
```sh
ERROR Error: Children must be not an empty array
    at TreeviewItem.set [as children] (treeview-item.js:109)
    at new TreeviewItem (
```

I believe this should work; instead of giving error. Can anyone help in this scenario!

**Comment by leovo2708** [comment]
Ok, I will fix this.

**Comment by mumairofficial** [comment]
@leovo2708 one thing more `(selectedChange)="onSelectedChange($event)"`  always return null instead of value

**Comment by leovo2708** [comment]
Please check my example. It's working. I wrote unit test with coverage > 95%, so I make sure it's ok.

**Comment by leovo2708** [comment]
Published in new version 1.2.1.
